src/overlays/actors/ovl_En_And/z_en_and.c

```diff
@@ -30,22 +29,154 @@ ActorInit En_And_InitVars = {
     (ActorFunc)EnAnd_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_0600B380;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/func_80C18B90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/func_80C18BD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/func_80C18C50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/EnAnd_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/EnAnd_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/EnAnd_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/func_80C18ED0.s")
+static AnimationInfoS sAnimationInfo[8] = {
+    { &gAndStaticAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAndIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAndWalkAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAndRaiseHeadAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },
+    { &gAndRaisedHeadLoopAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAndRaiseHandAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },
+    { &gAndRaisedHandLoopAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAndRaisedHandWalkAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_And/EnAnd_Draw.s")
+s32 EnAnd_ChangeAnim(EnAnd* this, s32 animIndex) {
+    s32 ret = false;
+
+    if (this->animIndex != animIndex) {
+        this->animIndex = animIndex;
+        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, animIndex);
+    }
+    return ret;
+}
+
+void EnAnd_Blink(EnAnd* this) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex >= EYE_TEXTURES_COUNT) {
+            this->blinkTimer = Rand_S16Offset(30, 30);
+            this->eyeTexIndex = 0;
+        }
+    }
+}
+
+void EnAnd_HandleCsAction(EnAnd* this, PlayState* play) {
+    s32 actionAnimations[] = { 0, 1, 2, 3, 5, 7 };
+    u16 csAction;
+    s32 actionIndex;
+
+    if (play->csCtx.state != CS_STATE_0) {
+        if (!this->hasAction) {
+            this->action = 0xFF;
+            this->hasAction = true;
+            this->prevAnimIndex = this->animIndex;
+        }
+        if (Cutscene_CheckActorAction(play, 0x235)) {
+            actionIndex = Cutscene_GetActorActionIndex(play, 0x235);
+            csAction = play->csCtx.actorActions[actionIndex]->action;
+            if (this->action != (u8)csAction) {
+                this->action = csAction;
+                EnAnd_ChangeAnim(this, actionAnimations[csAction]);
+            }
+            switch (this->action) {
+                case 3:
+                case 4:
+                    if ((this->animIndex == 3) || (this->animIndex == 5)) {
+                        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+                            EnAnd_ChangeAnim(this, this->animIndex + 1);
+                        }
+                    }
+                    break;
+            }
```

Comment:
```suggestion
                    }
                    break;

                default:
                    break;
            }
```
It's our style to always add a default to switches if it matches